src/Autofac/Core/Resolving/Pipeline/PipelineBuilderEnumerator.cs

```diff
@@ -39,16 +39,16 @@ public bool MoveNext()
                 return false;
             }
 
-            if (_current is object)
+            if (_current is not null)
             {
                 _current = _current.Next;
 
-                _ended = !(_current is object);
+                _ended = !(_current is not null);
```

Comment:
Should this not be `_ended = _current is null`?

Reply:
Uh. Y... yes? 😆  Fixed.